### PR TITLE
Adding check for zero length string

### PR DESCRIPTION
**Pull request by CoryCharlton**
## Description
- Adding check for zero length string in `string.IndexOf` and `string.LastIndexOf`

## Motivation and Context
- Fixes nanoFramework/Home#1404

## How Has This Been Tested?
- Unit tests -> https://github.com/CoryCharlton/nanoFramework.String.UnitTests
 
## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [X] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [X] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [X] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [X] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

**Comment by CoryCharlton** [comment]
I wanted to add unit tests for this to the `nanoFramework.CoreLibrary` project but I couldn't get the existing projects to run in my environment. So in the meantime I created some unit tests here: https://github.com/CoryCharlton/nanoFramework.String.UnitTests

File: src/CLR/CorLib/corlib_native_System_String.cpp [hunk 1]
```diff
@@ -558,85 +556,96 @@ HRESULT Library_corlib_native_System_String::FromCharArray( CLR_RT_StackFrame& s
...
+        inputIterator.SetInputUTF8(pString);
         searchLen = inputIterator.CountNumberOfCharacters();
     }
 
     // calculate input string length
-    inputIterator.SetInputUTF8( szText );
+    inputIterator.SetInputUTF8(szText);
     inputLen = inputIterator.CountNumberOfCharacters();
 
+    // If input is string.Empty bail early
```
Comment (CoryCharlton) [inline src/CLR/CorLib/corlib_native_System_String.cpp:633]: So so it doesn't get lost in all the code format changes this is my actual change :D

Comment (josesimoes) [reply src/CLR/CorLib/corlib_native_System_String.cpp:633]: Let me help you there. I'll do a code format fix, then you can rebased. That way your changes will be the only one.
(if this happens in the future please ping me so I can do the same 😉 )

**Comment by josesimoes** [comment]
> I wanted to add unit tests for this to the `nanoFramework.CoreLibrary` project but I couldn't get the existing projects to run in my environment. So in the meantime I created some unit tests here: https://github.com/CoryCharlton/nanoFramework.String.UnitTests


Defenetly better to add those.
Do this please: 
1) Submit PR against mscorlib. Tests will fail.
2) We get this one merged.
3) After new nanoCLR release is pushed, rerun the Unit tests and those will pass.

**Review by josesimoes — APPROVED** [review]
Nice catch! Thank you for the fix.

**Pull request merged**